### ISSUE TITLE
Thyra::get_Epetra_MultiVector hangs in debug build if some (not all) procs have 0 rows (round 2)

### ISSUE THREAD

**Issue by mhoemmen** [trilinos/Trilinos#2061]
Lines 799-807 of thyra/adapters/epetra/src/Thyra_EpetraThyraWrappers.cpp has the same bug that we found and fixed in #1941.

https://github.com/trilinos/Trilinos/blob/678c5eecbee6b565bc91c64ba0b3b0dd5e6bf0d3/packages/thyra/adapters/epetra/src/Thyra_EpetraThyraWrappers.cpp#L799

More specifically, we fixed the following (const) overload of `Thyra::get_Epetra_MultiVector` in #1941:
```
Teuchos::RCP<const Epetra_MultiVector>
Thyra::get_Epetra_MultiVector(
  const Epetra_Map &map,
  const MultiVectorBase<double> &mv
  );
```
Now we need to fix the following (nonconst) overload:
```
Teuchos::RCP<Epetra_MultiVector>
Thyra::get_Epetra_MultiVector(
  const Epetra_Map &map,
  MultiVectorBase<double> &mv
  );
```
The overloads are similar enough that we should be able to change the existing #1941 unit test to exercise the second (nonconst) overload as well.

@trilinos/thyra 

## Expectations

`Thyra::get_Epetra_MultiVector` needs not to hang, even if some (but not all) processes have zero rows.

## Current Behavior

`Thyra::get_Epetra_MultiVector` hangs, even if some (but not all) processes have zero rows.

## Motivation and Context

Our application uses Thyra sometimes to drive linear solvers.

## Related Issues

* Related to #1941

**Comment by mhoemmen** [comment]
I fixed this issue, and modified the #1941 unit test so that it exercises both the const and nonconst overloads of `Thyra::get_Epetra_MultiVector`.  I also improved the unit test so it ensures consistency over all MPI processes.  The unit test now passes locally.  I will run the remote check-in test script to push the fix to develop.

**Comment by mhoemmen** [comment]
Thanks all! :D
